src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -48,24 +50,70 @@ internal class IssuesSnapshot : WpfTableEntriesSnapshotBase
         private readonly IList<IssueMarker> issueMarkers;
         private readonly IReadOnlyCollection<IssueMarker> readonlyIssueMarkers;
 
+        /// <summary>
+        /// Logical identifier for a set of issues produced by an analysis run.
+        /// Every snapshot that has the same analysis id describes the same set of
+        /// issues in the same order. The implementation of "IndexOf" depends on this.
+        /// </summary>
+        public Guid AnalysisRunId { get; }
+
         public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers;
 
-        internal IssuesSnapshot(string projectName, string filePath, int versionNumber, IEnumerable<IssueMarker> issueMarkers)
-            : this(projectName, filePath, versionNumber, issueMarkers, new AnalysisSeverityToVsSeverityConverter(), new RuleHelpLinkProvider())
+        // Every snapshot has a unique version number. It doesn't matter what it is, as
+        // long as it increments (if two snapshots have the same number or lower, the ErrorList
+        // will assume the data hasn't changed and won't update the rows).
+        private static int nextVersionNumber = 0;
+        private static int GetNextVersionNumber() => ++nextVersionNumber;
+
+        #region Construction methods
+
+        /// <summary>
+        /// Create a snapshot with new set of issues from a new analysis run
+        /// </summary>
+        public static IssuesSnapshot CreateNew(string projectName, string filePath, IEnumerable<IssueMarker> issueMarkers) =>
```

Comment:
I've added three methods for the higher-level operations we want to perform. Each of those methods set the `AnalysisRunId` appropriately, depending on whether the snapshot relates to a different set of issues or the same set of locations with modified location spans.

If we could reliably (and cheaply) compare two lists of issue markers then we could automatically work out which operation to perform. However, this solution is much simpler and less risky.